### PR TITLE
Update the aggregated aww data based on last 6 month historical data

### PR DESCRIPTION
**Pull request by Rohit25negi**
http://manage.dimagi.com/default.asp?278126

### CODE REVIEW

**Comment by Rohit25negi** [comment]
@emord  please review this change about using the hh_num for launched awcs

**Review by emord — COMMENTED** [review]
I see now why we didn't/can't use the hh_num case property for this. For this indicator we want to be able to answer "In month X, did an AWC have any households?" Case properties only provide the current status, so the only question that would be able to answer is "Does this AWC have any households now?". If we wanted to be able to answer the original question with a case property we would need to have a case property that stored the date of first household registration.

Currently, to get that kind of historical data, we need to view the forms which is what we are doing now. I think we can fix the issue in the ticket in two possible ways:

1) Add a usage_table query with a where clause is < current month. Something like `SELECT count(*) > 1 FROM usage table where month < 1 and register_household_form = true;` However, I think that table is likely too big to do that.

2) Instead I think we should just add a new task that runs once a week. This task will run something like [these lines](https://github.com/dimagi/commcare-hq/blob/4a875f6c0326ac3dd8a80d1c3d69c569a492975a/custom/icds_reports/migrations/sql_templates/database_functions/aggregate_awc_data.sql#L207-L267) and their [subsequent rollup queries](https://github.com/dimagi/commcare-hq/blob/4a875f6c0326ac3dd8a80d1c3d69c569a492975a/custom/icds_reports/migrations/sql_templates/database_functions/aggregate_awc_data.sql#L462), and will update the changed AWCs

**Comment by Rohit25negi** [comment]
@emord  the problem in the ticket caused because the user's submitted the form much later than they have filled out the form and when they have submitted the `date of their filling` form is considered whether it lies in aggregation timespan(2 month) not the date of submission? Can we instead use the `date of submission` for this? I think if we can, it although won't correct the past data but this will prevent this happening in future without much calculation.

**Comment by Rohit25negi** [comment]
Hi @emord , I have added the weekly aggregation sql script.
This script will update the aggregated data based on previous 6 months. please check

**Comment by Rohit25negi** [comment]
considering [update works for inherited table.](https://www.postgresql.org/docs/9.5/static/ddl-inherit.html). Added sql script should achieve the what we desire and discussed.

File: custom/icds_reports/tasks.py
Comment (calellowitz) [inline custom/icds_reports/tasks.py]: i think this should have two blank lines (not sure why stickler didnt catch it)

**Review by calellowitz — APPROVED** [review]
One minor nit, but looks good. Will wait for @emord for final approval

**Comment by calellowitz** [comment]
also removed that label since this is only for icds, not all users

**Review by emord — APPROVED** [review]

**Pull request merged**
